src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTaskLogic.cs

```diff
@@ -39,7 +44,15 @@ public PackArgs GetPackArgs(IPackTaskRequest<IMSBuildItem> request)
                 PackTargetArgs = new MSBuildPackTargetArgs()
             };
 
-            packArgs.Logger = new PackCollectorLogger(request.Logger, packArgs.WarningProperties);
+            WarningPropertiesCollection warningPropertiesCollection = null;
+
+            if (_packageBuilder?.PackageSpecificNoWarnProperties.Keys.Count > 0)
```

Comment:
Should the package builder be passed in as an argument instead? 

If I'm reading it correctly, if you don't call GetPackageBuilder before GetPackArgs, you won't get the collection.

Reply:
Yes, if you don't call GetPackageBuilder before GetPackArgs, you won't get the collection.

Reply:
We pass as argument problem is it'll introduce Api changes, not sure if it's necessary.

Reply:
That shouldn't be a concern with classes in NuGet.Build.Tasks.Pack and NuGet.Build.Tasks. 

@zivkan I wasn't able to find an issue, but I remember we've discussed removing the analyzers from those 2 assemblies in the past? Or at the minimum shifting *all* classes that are a Task to internal. 

I know we did that for NuGet.Commandline.Xplat, should we do the same thing here?

Reply:
@zivkan 
Could you be able to take a look at above comment?

Reply:
It makes sense to remove the public api analyzer from `Nuget.Build.Tasks.Pack`. The nupkg the project creates doesn't have any assemblies under `lib/` or `ref/`, so anyone referencing the package can't reference APIs.  I'm not sure if making the classes and properties internal will work, since MSBuild needs to create and set properties on classes referenced by `<UsingTask`.

`NuGet.Build.Tasks`, unfortunately, is packed in a way where customers can reference the package in their project and they'll get both compile and runtime assets, so can theoretically call it as if it's an API. Ideally we'd change how it's packed to be more like the pack targets, but doing so risks breaking our insertions into the .NET SDK, and the effort to make this change would be hard to prioritize over other work.

In any case, as Nikolche said, it should not be a problem in theory. Customers should not be using either of these packages as an API, they are packages to enable features.

Reply:
@nkolev92 
Addressed, now I'm passing `WarningPropertiesCollection` as additional parameter for `GetPackArgs` method.